docs/django/image_field.md

```diff
@@ -0,0 +1,147 @@
+# Django Image field
+
+For the ImageField to work it's necessary to have the Pillow library installed.
+
+## Django Models
+
+The setup of the model containing the image field can be done in the following way.
+
+````python
+class BookInformation(models.Model):
+    #...
+    cover_image = models.ImageField(upload_to="covers/", null=True)
+    #...
+````
+Note that the `upload_to` option corresponds to the path in which the images are stored/saved in the media folder.
+In this case the path will be `.../library_project/media/covers/`.
+
+## Factory Boy mock data
+
+The easiest way to generate mock images data with factory boy is with the `factory.django.ImageField()` method just like the example below.
+
+```` python 
+class BookInformationFactory(factory.django.DjangoModelFactory):
+    #...
+    cover_image = factory.django.ImageField(color=factory.Faker("color"))
+    #...
+````
+
+Note that the `color` option can Hardcoded with something like `color="blue"` and that will result in the 
+generation of images with the color blue. In this case, `factory.Faker("color")` will generate random color names
+which will result in the generation of images with diferent colors for each `cover_image`.
+
+## Django Model Serializer 
+
+Considering the previously shown factory, here is the correspondent serializer for POST and PUT requests.  
+
+````python 
+class BookInformationCreateUpdateSerializer(serializers.ModelSerializer):
+    cover_image = serializers.ImageField(required=False, allow_empty_file=True)
+
+    class Meta:
+        model = BookInformation
+        fields = [     
+        #... , 
+        "cover_image", 
+        #... ,
+        ]
+````
+
+## API View
+
+````python
+class BookInformationViewset(viewsets.ModelViewSet):
+    queryset = BookInformation.objects.all()
+    serializer_class = BookInformationCreateUpdateSerializer
+````
+
+## ImageField Views Tests
+
+To test the views that involve the `cover_image` attribute you can simply generate an image based on the
+implemented factory for the cover image just like the example below.
+
+````python
+from PIL import Image
+import shutil
+from ..fixtures import REMOVE_MEDIA_FILES_PATH
+
+class BookCreateViewTestCase(APITestCase):
+    def setUp(self):
+        #...
+        self.data = {
+            "cover_image": #...
+            #...
+        }
+
+    def tearDown(self):
+        shutil.rmtree(REMOVE_MEDIA_FILES_PATH)
+
+    def test_create_with_image(self):
+        cover = BookInformationFactory.create().cover_image
+
+        self.data["cover_image"] = cover
+        res = self.client.post(self.url, self.data, format="multipart")
+
+        self.assertEqual(res.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(
+            BookInformation.objects.first().cover_image.read(), BookInformation.objects.last().cover_image.read()
```

Comment:
why are you comparing the first with the last? isn't it just 1 result?

Reply:
The first image inserted was stored in the ``cover`` variable. Then we will insert another row in the DB with ``self.client.post(...)`` with the same image. In the test we compare both instances (since the image must be the same).